metastore/validator.go

```diff
@@ -0,0 +1,193 @@
+package metastore
+
+import (
+	"github.com/uber/aresdb/metastore/common"
+	memCom "github.com/uber/aresdb/memstore/common"
+	"reflect"
+	"github.com/uber/aresdb/utils"
+)
+
+// TableSchemaValidator validates it a new table schema is valid, given existing schema
+type TableSchemaValidator interface {
+	Validate() error
+}
+
+// NewTableSchameValidator returns a new TableSchemaValidator. Pass nil for oldTable if none exists
+func NewTableSchameValidator(newTable *common.Table, oldTable *common.Table) TableSchemaValidator {
+	return tableSchemaValidatorImpl{
+		newTable: newTable,
+		oldTable: oldTable,
+	}
+}
+
+type tableSchemaValidatorImpl struct {
+	newTable *common.Table
+	oldTable *common.Table
+}
+
+func (v tableSchemaValidatorImpl) Validate() (err error) {
+	if v.oldTable == nil {
+		return v.validateIndividualSchema(v.newTable, true)
+	}
+	return v.validateSchemaUpdate(v.newTable, v.oldTable)
+}
+
+// checks performed:
+//	table has at least 1 valid column
```

Comment:
need to add checks:
1. achiving sort columns cannot have duplicate columnID
2. primary key colunns cannot have duplicate columnID
3. column name cannot duplicate